bundles/org.openhab.binding.proteusecometer/src/main/java/org/openhab/binding/proteusecometer/internal/ecometers/handler/ProteusEcoMeterSHandler.java

```diff
@@ -0,0 +1,157 @@
+ */
+package org.openhab.binding.proteusecometer.internal.ecometers.handler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.proteusecometer.internal.ProteusEcoMeterBindingConstants;
+import org.openhab.binding.proteusecometer.internal.ProteusEcoMeterConfiguration;
+import org.openhab.binding.proteusecometer.internal.WrappedException;
+import org.openhab.binding.proteusecometer.internal.ecometers.ProteusEcoMeterSReply;
+import org.openhab.binding.proteusecometer.internal.ecometers.ProteusEcoMeterSService;
+import org.openhab.binding.proteusecometer.internal.serialport.SerialPortService;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fazecast.jSerialComm.SerialPort;
+
+/**
+ * The {@link ProteusEcoMeterSHandler} updates thing channels when receiving data
+ *
+ * @author Matthias Herrmann - Initial contribution
+ */
+@NonNullByDefault
+public class ProteusEcoMeterSHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ProteusEcoMeterSHandler.class);
+    private @Nullable SerialPort serialPort;
+    private @Nullable ProteusEcoMeterConfiguration config;
+    private @Nullable ScheduledFuture<?> job;
+
+    public ProteusEcoMeterSHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ProteusEcoMeterConfiguration.class);
+        final ProteusEcoMeterConfiguration localConfig = config;
+        if (localConfig == null) {
+            final String msg = "Expected initialize to be called ONLY when config is not null";
+            logger.error(msg);
+            throw new IllegalStateException(msg);
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+        job = scheduler.schedule(() -> {
+
+            try {
+                final ProteusEcoMeterSService ecoMeterSService = new ProteusEcoMeterSService();
+                final Stream<ProteusEcoMeterSReply> replyStream = ecoMeterSService.read(localConfig.usbPort,
+                        serialPortService);
+                updateStatus(ThingStatus.ONLINE);
+
+                replyStream.forEach(reply -> {
+                    updateState(ProteusEcoMeterBindingConstants.SENSOR_LEVEL,
```

Comment:
```suggestion
                    updateState(SENSOR_LEVEL,
```
Use an import at the top and you can remove all of these. Please do to all of them.